### PR TITLE
[iOS] Unit Test Logging Update

### PR DESCRIPTION
**Pull request by jwoo-msft**
# Related Issue
reference #6762 
# Description
* output from printf is visible in Xcode, but it's still get stripped in the log of pipeline.
* Updated unit test with new unit test logging introduced in ios 12.

# How Verified
Tested with json with syntax errors.
 
###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/AdaptiveCards/pull/6844)

### CODE REVIEW

File: source/ios/AdaptiveCards/ADCIOSVisualizer/ADCIOSVisualizerTests/ADCIOSVisualizerTests.mm [hunk 1]
```diff
@@ -200,35 +200,47 @@ - (void)parseAndRenderDFS:(NSString *)rootPath FileManager:(NSFileManager *)fMan
...
-            for (NSError *parseError in cardParseResult.parseErrors) {
-                printf("%s\n", [parseError.userInfo[NSLocalizedDescriptionKey] UTF8String]);
-            }
-            XCTAssertTrue(NO);
+            [self assertParsing:cardParseResult fileName:fileName];
+        }
+    }
+}
+
+- (void)assertParsing:(ACOAdaptiveCardParseResult *)parseResult fileName:(NSString *)fileName
```
[the remark this answers is not in the document]
Comment (jwoo-msft) [reply source/ios/AdaptiveCards/ADCIOSVisualizer/ADCIOSVisualizerTests/ADCIOSVisualizerTests.mm:224]: are you referring to `(ACOAdaptiveCardParseResult *)parseResult`?
or to `assertParsing`?

**Review by almedina-ms — APPROVED** [review]
:shipit:

**Review by licanhua — APPROVED** [review]

File: source/ios/AdaptiveCards/ADCIOSVisualizer/ADCIOSVisualizerTests/ADCIOSVisualizerTests.mm [hunk 2]
```diff
@@ -200,35 +200,47 @@ - (void)parseAndRenderDFS:(NSString *)rootPath FileManager:(NSFileManager *)fMan
...
+{
+    if (!parseResult.isValid) {
+        NSMutableArray<NSString *> *errorMsgs = [[NSMutableArray alloc] init];
```
Comment (licanhua) [inline source/ios/AdaptiveCards/ADCIOSVisualizer/ADCIOSVisualizerTests/ADCIOSVisualizerTests.mm:227]: should you call [errorMsgs release]?

Comment (jwoo-msft) [reply source/ios/AdaptiveCards/ADCIOSVisualizer/ADCIOSVisualizerTests/ADCIOSVisualizerTests.mm:227]: no, we don't.

**Pull request merged**